FWCore/Framework/interface/Event.h

```diff
@@ -611,5 +611,37 @@ namespace edm {
     Handle<View<ELEMENT> > h(&*newview, bh.provenance());
     result.swap(h);
   }
+
+  // Free functions to retrieve a collection from the Event.
+  // Will throw an exception if the collection is not available.
+
+  template <typename T>
+  T const& get(Event const& event, InputTag const& tag) {
+    Handle<T> handle;
+    event.getByLabel(tag, handle);
+    if (not handle.isValid())
+      throw * handle.whyFailed();
+    return * handle.product();
+  }
+
+  template <typename T>
+  T const& get(Event const& event, EDGetToken const& token) {
+    Handle<T> handle;
+    event.getByToken(token, handle);
+    if (not handle.isValid())
```

Comment:
This check is not needed. Calling `handle.product()` will already throw the exception.
See
https://github.com/cms-sw/cmssw/blob/master/DataFormats/Common/src/HandleBase.cc#L7

So all calls using `Handle` do not need the check.

Reply:
Right, fixed.